test/blackbox-tests/test-cases/dune-cache/trim.t

```diff
@@ -196,11 +196,7 @@ it turned out to be flaky so we subsequently deleted it in #4511.
 Test the error message when using removed subcommands [start] and [stop].
 
   $ dune cache start
-  Error: Dune no longer uses the cache daemon, and so the `start` and `stop`
-  subcommands of `dune cache` were removed.
-  [1]
+  dune: command 'start': This subcommand is no longer supported and has been removed.
```

Comment:
This is the deprecated feature of cmdliner. TBH I am not really fan of just printing a string here. cc @emillon I wonder if it would be possible to replace this with something generic so that we can use our own error message printing?

Reply:
Do you want `dune cache start` to call into `User_error.raise` for example? You can do that with a "normal" command. The pattern if you want a command with no argument is:

```ocaml
let+ () = Term.const () in
User_error.raise ...
```

(there's an example in `bin/internal.ml`)

Reply:
Ah of course. Silly me.